### PR TITLE
Added ability to specify a LoginId field when creating new CognitoIdentity

### PR DESCRIPTION
**Pull request by chrisradek**
@AdityaManohar @jeskew

### CODE REVIEW

File: test/credentials.spec.coffee [hunk 1]
```diff
@@ -832,3 +840,26 @@ describe 'AWS.CognitoIdentityCredentials', ->
...
+              SecretAccessKey: 'SECRET'
+              SessionToken: 'TOKEN'
+          cb null
+        creds.refresh(->)
+        creds.getId (err, id) ->
+          expect(id).to.equal('IDENTITY-ID2')
+          expect(creds.getStorage('id')).to.equal('IDENTITY-ID2')
+        setupCreds({LoginId: 'LOGINIDB'})
+        expect(creds.getStorage('id')).not.to.equal('IDENTITY-ID2')
+        setupCreds({LoginId: 'LOGINIDA'})
```
Comment (jeskew) [inline test/credentials.spec.coffee]: Should there be another `expect` check after this line?

Comment (chrisradek) [inline test/credentials.spec.coffee]: These last 2 steps are actually a clean-up step.  When testing in a browser environment, these credentials would be cached, so the tests would pass the first time it was run in a browser, but not subsequent times. I could explicitly delete the credentials form localstorage instead, so it's more clear why I'm doing this.

Comment (jeskew) [inline test/credentials.spec.coffee]: OIC; I thought that's what the next line was doing.

**Comment by chrisradek** [comment]
I added a minor simplification so it is hopefully clearer which LoginId is getting deleted when the unit test completes.

**Comment by jeskew** [comment]
:shipit:

**Pull request merged**